pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -1743,7 +1748,7 @@ private void resendMessages(ClientCnx cnx, long expectedEpoch) {
                     cnx.channel().close();
                     return;
                 }
-                int messagesToResend = pendingMessages.size();
+                long messagesToResend = pendingMessages.messagesCount();
```

Comment:
Nit: technically, users might notice a change in the `Re-Sending` log line below when running with batch messaging enabled. I don't think this is a problem, but I wanted to mention the behavior change.

Reply:
Yes, though the message is more accurate now compared to before :)